java/code/src/com/redhat/rhn/frontend/xmlrpc/system/SystemHandler.java

```diff
@@ -1536,10 +1538,11 @@ public String downloadSystemId(User loggedInUser, Integer sid) throws FaultExcep
      * @xmlrpc.returntype
      *      #array_begin()
      *          #struct_begin("package")
+     *                 #prop("int", "package_id")
```

Comment:
suggestion: you could explain under which conditions the package id is `-1`. I think that happenes for packages which are installed, but not available in Uyuni/SUSE Manager.